mkdocs/docs/HPC/setting_up_python_virtual_environments.md

```diff
@@ -0,0 +1,262 @@
+import torch
+import poutyne
+
+...
+```
+
+We load a PyTorch package as a module and install Poutyne in a virtual environment:
+
+```bash
+$ module load PyTorch/2.1.2-foss-2023a
+$ python -m venv myenv
+$ source myenv/bin/activate
+$ pip install Poutyne
+```
+
+While the virtual environment is activated, we can run the script without any issues:
+
+```bash
+$ python pytorch_poutyne.py
+```
+
+
+## Creating a virtual environment for a specific cluster
+
+To create a virtual environment for a specific cluster, you need to start an interactive shell on that cluster.
+Let's say you want to create a virtual environment on the `donphan` cluster.
+
+```bash
+$ module swap cluster/donphan
+$ qsub -I
+```
+
+After some time, a shell will be started on the `donphan` cluster. 
+You can now create a virtual environment as described in [the first section](#creating-a-venv).
+This virtual environment can be used by jobs running on the `donphan` cluster.
+
+
+## Example Python job
+
+This section will combine the concepts discussed in the previous sections to:
+
+1. Create a virtual environment on a specific cluster.
+2. Combine packages installed in the virtual environment with modules.
+3. Submit a job script that uses the virtual environment.
+
+The example script that we will run is the following:
+
+```python title="pytorch_poutyne.py"
+import torch
+import poutyne
+
+print(f"The version of PyTorch is: {torch.__version__}")
+print(f"The version of Poutyne is: {poutyne.__version__}")
+```
+
+First, we create a virtual environment on the `donphan` cluster:
+
+```bash
+$ module swap cluster/donphan
+$ qsub -I
+# Load module dependencies
+$ module load PyTorch/2.1.2-foss-2023a
+$ python -m venv myenv
+$ source myenv/bin/activate
+# install virtual environment dependencies
+$ pip install Poutyne
+$ deactivate
+```
+
+We exit the interactive shell by pressing `CTRL+D` and create a job script that loads the PyTorch module, 
+enters the virtual environment and executes the script:
+
+```bash title="jobscript.pbs"
+#!/bin/bash
+
+# Basic parameters
+#PBS -N python_job_example    ## Job name
+#PBS -l nodes=1:ppn=1         ## 1 node, 1 processors per node
+#PBS -l walltime=01:00:00     ## Max time your job will run (no more than 72:00:00)
+
+module load PyTorch/2.1.2-foss-2023a   # Load the PyTorch module
+
+cd $PBS_O_WORKDIR             # Change working directory to the location where the job was submitted
+
+source myenv/bin/activate     # Activate the virtual environment
+python pytorch_poutyne.py     # Run your Python script, or any other command within the virtual environment
+deactivate                    # Deactivate the virtual environment
+```
+
+Next, we submit the job script:
+
+```bash
+$ qsub jobscript.pbs
+```
+
+Two files will be created in the directory where the job was submitted: `python_job_example.o[job_id]` and `python_job_example.e[job_id]`.
+The `.o` file contains the output of the job.
+
+
+## Troubleshooting
+
+### Illegal instruction error
```

Comment:
Make a similar subsection for GLIBC errors which you'll get when running a venv across different major OS versions?

Reply:
When playing around with this I noticed this is not only the case with venvs, but with almost any command. The following also fails:

```bash
$ ml Python
$ ml swap cluster/shinx
$ ml swap cluster/doduo
/usr/bin/lua: /lib64/libc.so.6: version `GLIBC_2.33' not found (required by /apps/gent/RHEL9/zen4-ib/software/ncurses/6.5-GCCcore-13.3.0/lib/libncurses.so.6)
```

Reply:
This issue might be worth adding to the general troubleshooting page.

Reply:
Added in [add GLIBC troubleshooting section](https://github.com/hpcugent/vsc_user_docs/pull/639/commits/999b560119911946e3f2331fbfe60dc58e1fac19).

Reply:
> This issue might also be worth adding to the general troubleshooting page.

Yes, agreed, there's an entry in your TODO list for this